pkg/backoff/exponential.go

```diff
@@ -38,22 +45,53 @@ type ExponentialBackOff struct {
 }
 
 // NewExponentialBackOff creates an istio wrapped ExponentialBackOff.
+// By default, it never stops.
 func NewExponentialBackOff(initFuncs ...func(off *ExponentialBackOff)) BackOff {
-	b := &ExponentialBackOff{}
+	b := ExponentialBackOff{}
 	b.ExponentialBackOff = backoff.NewExponentialBackOff()
+	b.ExponentialBackOff.MaxElapsedTime = 0
 	for _, fn := range initFuncs {
-		fn(b)
+		fn(&b)
 	}
 	b.Reset()
 	return b
 }
 
 const MaxDuration = 1<<63 - 1
 
-func (b *ExponentialBackOff) NextBackOff() time.Duration {
+func (b ExponentialBackOff) NextBackOff() time.Duration {
 	duration := b.ExponentialBackOff.NextBackOff()
 	if duration == b.Stop {
 		return MaxDuration
```

Comment:
This seems very unsafe.. callers will `sleep(forever)` when this happens.

IMO all backoffs either need to explicitly fail when reaching max time or just be capped at max interval

Reply:
fixed